### PR TITLE
Adding SDK for litmus-python and updating base code changes

### PR DESCRIPTION
**Pull request by oumkale**
Signed-off-by: Oum Kale <oumkale@chaosnative.com>

SDK for litmus-python added
  - Run the following command to generate the necessary artifacts for submitting the `samplecategory` chaos chart.

  ```
  $ python3 generate_experiment.py -f=attributes.yaml -g=<generate_type>
  ```

  **Note**: Replace the `generate_type=<generate-type>` placeholder with the appropriate value based on the usecase: 
  - `experiment`: Chaos experiment artifacts belonging to an existing OR new experiment.
  - `chart`: Just the chaos-chart metadata, i.e., experiment_chartserviceversion.yaml
      - Provide the type of chart in the `-t` flag. It supports the following values:
           - `category`: It creates the chart metadata for the category i.e chartserviceversion, package manifests
           - `experiment`: It creates the chart for the experiment i.e chartserviceversion, engine, rbac, experiment manifests
           - `all`: it creates both category and experiment charts (default type)

### CODE REVIEW

File: pkg/utils/common/pods.py [hunk 1]
```diff
@@ -86,8 +86,8 @@ def FilterNonChaosPods(self, podList, chaosDetails, clients):
 			nonChaosPods = []
 			# ignore chaos pods
 			for pod in podList.items:
-				if (pod.metadata.labels["chaosUID"] != str(chaosDetails.ChaosUID) or pod.metadata.labels["name"] == "chaos-operator"):
-					nonChaosPods = nonChaosPods.append(pod)
+				if pod.metadata.labels.get("chaosUID") != str(chaosDetails.ChaosUID) or pod.metadata.labels.get("name") != "chaos-operator":
```
Comment (ispeakc0de) [inline pkg/utils/common/pods.py:89]:
```suggestion
				if pod.metadata.labels.get("chaosUID") == "" and pod.metadata.labels.get("name") != "chaos-operator":
```

File: pkg/utils/exec/exec.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return ValueError("%s pod is not in running state, phase: %s", pod.Name, pod.Status.Phase)
+
+    for container in pod.status.container_statuses:
+        if container.name == containerName and container.state.terminated != None:
+            return ValueError("%s container of %s pod is not in ready state, phase: %s", container.name, pod.metadata.name, pod.status.phase)
+
+    return None
+
+def Exec(commandDetails, clients , command):
+    name=commandDetails.PodName
```
Comment (ispeakc0de) [inline pkg/utils/exec/exec.py:24]: Do we need an extra variable here, can't we directly use commandDetails.PodName? -- as we are not modifying its value in this function

File: pkg/utils/exec/exec.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		self.ContainerName = ContainerName
+
+# checkPodStatus verify the status of given pod & container
+def checkPodStatus(pod , containerName):
+
+    if pod.status.phase.lower() != "running":
+        return ValueError("%s pod is not in running state, phase: %s", pod.Name, pod.Status.Phase)
+
+    for container in pod.status.container_statuses:
+        if container.name == containerName and container.state.terminated != None:
```
Comment (ispeakc0de) [inline pkg/utils/exec/exec.py:18]: why are we checking container state for termination? can't we use `!container.Ready`?

File: contribute/developer-guide/README.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
+## Steps to Bootstrap a Chaos Experiment
+
+The artifacts associated with a chaos-experiment are summarized below: 
+
+- Submitted in the litmuschaos/litmus-python repository, under the experiments/*chaoscategory*/*experiment-name* folder 
+
+  - Experiment business logic in golang. May involve creating new or reusing an existing chaosLib
```
Comment (ispeakc0de) [inline contribute/developer-guide/README.md:7]:
```suggestion
  - Experiment business logic in python. May involve creating new or reusing an existing chaosLib 
```

File: contribute/developer-guide/README.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+The artifacts associated with a chaos-experiment are summarized below: 
+
+- Submitted in the litmuschaos/litmus-python repository, under the experiments/*chaoscategory*/*experiment-name* folder 
+
+  - Experiment business logic in golang. May involve creating new or reusing an existing chaosLib 
+  - Experiment test deployment manifest that is used for verification purposes
+  - Experiment RBAC (holds experiment-specific ServiceAccount, Role and RoleBinding)
+
+  Example: [pod delete experiment in litmus-python](/experiments/generic/pod-delete)
```
Comment (ispeakc0de) [inline contribute/developer-guide/README.md:11]: will it redirect to the corresponding experiment?

File: contribute/developer-guide/README.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+### Steps to Generate Experiment Manifests
+
+- Clone the litmus-python repository & navigate to the `contribute/developer-guide` folder
+
+  ```
+  $ git clone https://github.com/litmuschaos/litmus-python.git
+  $ cd litmus/contribute/developer-guide
+  ```
+
+- Populate the `attributes.yaml` with details of the chaos experiment (or chart). Use the [attributes.yaml.sample](/contribute/developer-guide/attributes.yaml.sample) as reference.
```
Comment (ispeakc0de) [inline contribute/developer-guide/README.md:40]: provide the correct url?

File: contribute/developer-guide/README.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+  ```
+
+- Run the following command to generate the necessary artefacts for submitting the `samplecategory` chaos chart with 
+  `pod-delete` experiment.
+
+  ```
+  $ python3 generate_experiment.py --attributes_file=attributes.yaml --generate_type=experiment
+  ```
+
+  **Note**: Replace the `generate_type=<experiment>` placeholder with the appropriate value based on the usecase:
```
Comment (ispeakc0de) [inline contribute/developer-guide/README.md:108]:
```suggestion
  **Note**: Provide the `generate_type` flag based on the usecase: 
```

File: contribute/developer-guide/README.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+    - name: Documentation
+      url: "https://docs.litmuschaos.io/docs/getstarted/"
+
+  ```
+
+- Run the following command to generate the necessary artefacts for submitting the `samplecategory` chaos chart with 
+  `pod-delete` experiment.
+
+  ```
+  $ python3 generate_experiment.py --attributes_file=attributes.yaml --generate_type=experiment
```
Comment (ispeakc0de) [inline contribute/developer-guide/README.md:105]: Can we keep the same flag names as of litmus-go repo?

File: contribute/developer-guide/attributes.yaml [hunk 9]
```diff
@@ -0,0 +1,51 @@
+---
+name: "sample_exec_chaos"
+version: "0.1.0"
+category: "sample_category"
+repository: "https://github.com/litmuschaos/litmus-python/tree/master/sample-category/sample-exec-chaos"
```
Comment (ispeakc0de) [inline contribute/developer-guide/attributes.yaml:5]:
```suggestion
repository: "https://github.com/litmuschaos/litmus-python/tree/master/sample_category/sample_exec_chaos"
```

File: contribute/developer-guide/attributes.yaml.sample [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+description: "it execs inside target pods to run the chaos inject commands, waits for the chaos duration and reverts the chaos"
+keywords:
+  - "pods"
+  - "kubernetes"
+  - "sample-category"
+  - "exec"
+platforms:
+  - Minikube
+scope: "Namespaced"
+auxiliaryappcheck: true
```
Comment (ispeakc0de) [inline contribute/developer-guide/attributes.yaml.sample:16]:
```suggestion
auxiliaryappcheck:  false
```

File: contribute/developer-guide/README.md [hunk 11]
```diff
@@ -0,0 +1,308 @@
...
+    - name: Documentation
+      url: "https://docs.litmuschaos.io/docs/getstarted/"
+
+  ```
+
+- Run the following command to generate the necessary artefacts for submitting the `sample_category` chaos chart with 
+  `sample_exec_chaos` experiment.
+
+  ```
+  $ python3 generate_experiment.py -f=attributes.yaml -g=<experiment>
```
Comment (ispeakc0de) [inline contribute/developer-guide/README.md:105]:
```suggestion
  $ python3 generate_experiment.py -f=attributes.yaml -g=experiment
```

File: contribute/developer-guide/generate_experiment.py [hunk 12]
```diff
@@ -0,0 +1,274 @@
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+import yaml
+import os
+import sys
+import argparse
+import glob
+import shutil
+
+# generate_csv creates __init__.py file which makes python to treat directories containing it as modules
```
Comment (ispeakc0de) [inline contribute/developer-guide/generate_experiment.py:9]:
```suggestion
# generate_csv creates __init__.py file which helps python to find the corresponding module
```

**Review by ispeakc0de — APPROVED** [review]

**Review by uditgaurav — APPROVED** [review]

**Pull request merged**